addon/-private/system/store.js

```diff
@@ -2787,19 +2797,82 @@ function _commit(adapter, store, operation, snapshot) {
   }, label);
 }
 
-function setupRelationships(store, internalModel, data) {
+function inverseRelationshipInitialized(store, internalModel, data, key, modelNameToInverseMap) {
+  let relationshipData = data.relationships[key].data;
+
+  if (!relationshipData) {
+    // can't check inverse for eg { comments: { links: { related: URL }}}
+    return false;
+  }
+
+  let inverseMap = modelNameToInverseMap[internalModel.modelName]
+  if (!inverseMap) {
+    inverseMap = modelNameToInverseMap[internalModel.modelName] = get(internalModel.type, 'inverseMap');
+  }
+  let inverseData = inverseMap[key];
+  if (inverseData === undefined) {
```

Comment:
lets rename this to something like `inverseRelationshipMetadata`/something not containing the word `Data`